### ISSUE TITLE
terraform state show results in unsupported attribute "hibernation" error

### ISSUE THREAD

**Issue by namdets** [terraform-providers/terraform-provider-aws#12059]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.12.20
+ provider.aws v2.47.0
+ provider.local v1.4.0
+ provider.random v2.2.1
+ provider.template v2.1.2

I also tried using v0.12.16 which had the same result. 

### Affected Resource(s)

* aws_instance

### Terraform Configuration Files

```data "template_file" "hpds-user_data" {
  template = file("scripts/hpds-user_data.sh")
  vars = {
    stack_githash = var.stack_githash_long
    dataset_s3_object_key = var.dataset-s3-object-key
    stack_s3_bucket = var.stack_s3_bucket
  }
}

data "template_cloudinit_config" "hpds-user-data" {
  gzip          = true
  base64_encode = true

  # user_data
  part {
    content_type = "text/x-shellscript"
    content      = data.template_file.hpds-user_data.rendered
  }

}

resource "aws_instance" "hpds-ec2" {
  depends_on = [
    local_file.wildfly-standalone-xml-file
  ]

  ami = var.ami-id
  instance_type = "m5.xlarge"

  associate_public_ip_address = false

  subnet_id = var.db-subnet-us-east-1a-id

  iam_instance_profile = "hpds-deployment-s3-profile-${var.target-stack}-${var.stack_githash}"

  user_data = data.template_cloudinit_config.hpds-user-data.rendered

  vpc_security_group_ids = [
    aws_security_group.inbound-hpds-from-app.id,
    aws_security_group.outbound-to-trend-micro.id,
    aws_security_group.inbound-app-from-lma-for-dev-only.id
  ]
  root_block_device {
    delete_on_termination = true
    encrypted = true
    volume_size = 50
  }

  tags = {
    Owner       = "Avillach_Lab"
    Environment = "development"
    Name        = "FISMA Terraform Playground - ${var.stack_githash} - HPDS - ${var.target-stack}"
  }

}

resource "aws_route53_record" "hpds" {
  zone_id = var.internal-dns-zone-id
  name    = "hpds.${var.target-stack}"
  type    = "A"
  ttl     = "60"
  records = [aws_instance.hpds-ec2.private_ip]
}```

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

https://gist.github.com/namdets/07aa41c179749f9111e7737227d454a4

### Panic Output

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->

### Expected Behavior

<!--- What should have happened? --->
I should have seen the terraform state for the aws_instance.hpds-ec2 resource, which successfully has been created.

### Actual Behavior

<!--- What actually happened? --->

I see the following error and no other output:

unsupported attribute "hibernation"

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply`
2. `terraform state show aws_instance.hpds-ec2`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? --->

This problem did not exist last week. AWS may have updated their API.

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

* #0000

**Comment by namdets** [comment]
This may have been fixed by this commit: https://github.com/terraform-providers/terraform-provider-aws/commit/e65315b0a81367ea005b79ebcb4d9a7749991eec

I'll test and post back.

**Comment by namdets** [comment]
Not fixed in release 0.12.21

Not sure how to see if e65315b is in 0.12.21 or not

**Comment by bflad** [comment]
Hi @namdets 👋 Terraform CLI (version 0.12.21 in your case) is versioned and released separately from the Terraform AWS Provider (version 2.49.0 currently with 2.50.0 shipping later today). You'll want to ensure your Terraform AWS Provider version is 2.49.0 or later for `aws_instance` resource `hibernation` support. Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Comment by namdets** [comment]
Hi @bflad, Thanks for the info.

My CI installs terraform from scratch and all providers from there on each run.

This is from the latest one just now:

```
+ unzip terraform_0.12.21_linux_amd64.zip
Archive:  terraform_0.12.21_linux_amd64.zip
  inflating: terraform               
+ mv terraform /usr/local/bin/
+ terraform init

[0m[1mInitializing the backend...[0m

[0m[1mInitializing provider plugins...[0m

The following providers do not have any version constraints in configuration,
so the latest version was installed.

To prevent automatic upgrades to new major versions that may contain breaking
changes, it is recommended to add version = "..." constraints to the
corresponding provider blocks in configuration, with the constraint strings
suggested below.

* provider.aws: version = "~> 2.47"
* provider.local: version = "~> 1.4"
* provider.random: version = "~> 2.2"
* provider.template: version = "~> 2.1"
```

It says that the latest version of each provider was installed, but the aws provider is only version 2.47, is there something special I need to do to opt into versions past 2.49?

Thanks!

Jason

**Comment by namdets** [comment]
Ah, my mistake, the workspace clearing checkbox was unchecked so it was using the pre-existing version. I'll report back and close the issue if 2.49 fixes it.

Thanks!

**Comment by namdets** [comment]
Ok, I confirmed it is fixed in 2.49

Thanks!
